instrumentation/jaxrs/jaxrs-1.0/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxRsAnnotationsTracer.java

```diff
@@ -26,23 +27,24 @@
 
   public Span startSpan(Class<?> target, Method method) {
     String pathBasedSpanName = getPathSpanName(target, method);
-    Span serverSpan = BaseTracer.getCurrentServerSpan();
+    Context context = Context.current();
+    Span serverSpan = BaseTracer.getCurrentServerSpan(context);
 
     // When jax-rs is the root, we want to name using the path, otherwise use the class/method.
     String spanName;
     if (serverSpan == null) {
       spanName = pathBasedSpanName;
     } else {
       spanName = spanNameForMethod(target, method);
-      updateServerSpanName(serverSpan, pathBasedSpanName);
+      updateServerSpanName(context, serverSpan, pathBasedSpanName);
     }
 
     return tracer.spanBuilder(spanName).startSpan();
   }
 
-  private void updateServerSpanName(Span span, String spanName) {
+  private void updateServerSpanName(Context context, Span span, String spanName) {
     if (!spanName.isEmpty()) {
-      span.updateName(spanName);
+      span.updateName(BaseTracer.getApplicationRoot(context) + spanName);
```

Comment:
Also I don't like this scattered non-obvious logic :( Several places which are very far from each other and not really related now have to implement the same rule for span name. If/when we change them again, it is certain we don't find all these places.

At the very least we should encapsulate this name rule and application root access into a separate class. E.g. `ServletSpanName` or something.

Reply:
I think this is addressed now, let me know what u think